### PR TITLE
[network] add NetworkId to config to differentiate

### LINKED ISSUE #4012: [network] disambiguate networks with `NetworkId` to fix network metrics/logs.
Replaces prev. issue #2397

Relevant issue #3341

### Problem

It's painful to view prometheus metrics or node logs for network when multiple network instances are running, e.g., a validator is connected to the validator swarm via one instance and their fullnode swarm via another instance. The issue is that network metrics don't have an extra label/dimension for the network type to ensure that the values aren't accidentally combining. Similarly, most log entries in network are difficult to disambiguate across multiple network instances. For example, the Rpc "cancelled requests" metric doesn't have a label for the network type, so it accidentally counts the total number of cancelled requests across all network instances without any way to refine the query to a specific network instance.

A separate (but somewhat orthogonal) problem is that application protocols like state sync or mempool have trouble disambiguating between network messages from different network instances. To unblock those protocols, we recently added some quick changes to state sync/mempool to deal with ambiguous network messages (#3568). However, we should probably find a long-term solution to this issue in network itself.

### Solution

Since different network instances are currently ambiguous, we can add a `NetworkId` to uniquely identify each one. This type will probably look something like `RoleType`. Unfortunately, we can't use `RoleType` to disambiguate networks as the existing `enum RoleType { Validator, FullNode }` describes a specific node's role inside a network instance, rather than the network instance itself.

Instead we can probably get by with exhaustively enumerating the network instance types in `NetworkId`:

```rust
// should probably look something like RoleType:
// https://github.com/libra/libra/blob/master/config/src/config/mod.rs#L110

#[repr(u8)]
enum NetworkType {
    Validator = 0,
    Public,
    Private,
}
struct NetworkId {
    network_type: NetworkType,
    network_name: String
    // ..
}
```

+ Each `NetworkConfig` should have a `network_id`. (https://github.com/libra/libra/blob/master/config/src/config/network_config.rs#L26)

+ `config-builder` should emit the right `network_id` for the `NetworkConfig`s it generates.
(https://github.com/libra/libra/blob/master/config/config-builder/src/validator_config.rs, https://github.com/libra/libra/blob/master/config/config-builder/src/full_node_config.rs, https://github.com/libra/libra/blob/master/config/src/generator.rs)

+ Will need some sanity checks in `NodeConfig` to check we don't have multiple `NetworkConfig`s with the same `NetworkId`.

+ `NetworkBuilder::new(..)` should probably take a `network_id`. (https://github.com/libra/libra/blob/master/network/src/validator_network/network_builder.rs#L148)

+ All network counters/metrics should include a new `network_id` label. We probably need to push down `NetworkId` into all network actors (peer_manager, peer, rpc, direct_send, connectivity_manager, ..)

+ All logs in network should include appropriate context (`network_id`, `role_type`, `peer_id`).

+ Later, we could also use this `NetworkId` to solve state sync/mempool issues with ambiguous network events. For example, we could tag `Event<TMessage>` (https://github.com/libra/libra/blob/master/network/src/protocols/network/mod.rs) with `NetworkId`. (cc @christinacelee would this actually be helpful?).

+ Add `NetworkId` (and `ChainId` #4011) to handshake message (https://github.com/libra/libra/blob/master/network/src/protocols/wire/handshake/v1/mod.rs#L107). Should help prevent accidental network discovery collisions.

### PR DESCRIPTION
**Pull request by gregnazario**
## Motivation

Each network device that is on a different network currently all log
the same way, which makes it difficult to determine which network that
one is looking at.  This allows us to differentiate both at a logging
level and later at a communication level between networks.

resolves #4012

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Added tests to check that network handshakes work with the new network Ids.  This also means that both sides need to be updated for communication to work.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: types/src/network_id.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// SPDX-License-Identifier: Apache-2.0
+use serde::{Deserialize, Serialize};
+
+/// A representation of the network being used in communication.
+/// There should only be one of each NetworkId used for a single node, and handshakes should verify
+/// that the NetworkId being used is the same during a handshake, to effectively ensure communication
+/// is restricted to a network.  Network should be checked that it is not the `DEFAULT_NETWORK`
+/// @TODO: Figure out why a normal struct with an enum and String doesn't work here for serialization
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct NetworkId(String);
```
Comment (davidiw) [inline types/src/network_id.rs:11]: PR is coming along really well! Thought I had left a comment somewhere. I left a note in the issue. I think we want an enum here, it will make it much more flexible / usable going forward. Also I think it is okay for this to live in the config itself, was there a reason to have it types?

Comment (gregnazario) [reply types/src/network_id.rs:11]: Wasn't quite sure where to put it so I put it where PeerId / AccountAddress was.  I did get a message about the enum, and planning to do it next :)  The Newtype struct was a stopgap to get the tests to run and not crash on previous attempts of an enum that didn't serialize to toml.

If its only used for config/handshake purposes i think config is fine.  if we use it past just handshakes, we might want to leave it in the types to be more general

Comment (davidiw) [reply types/src/network_id.rs:11]: Sweet. In general, types is intended to be our dumping ground for client interfaces. The network id is probably outside the scope of that at least at this time , /I think/.

Comment (gregnazario) [reply types/src/network_id.rs]: Sounds good, moved to config.

File: config/config-builder/src/full_node_config.rs [hunk 2]
```diff
@@ -166,6 +168,7 @@ impl FullNodeConfig {
         let new_net = full_node_config.full_node_networks.swap_remove(0);
         for net in &config.full_node_networks {
             ensure!(new_net.peer_id != net.peer_id, "Network already exists");
+            // TODO do we want to distinguish between public network ids
```
Comment (davidiw) [inline config/config-builder/src/full_node_config.rs:171]: There is only one public network, in my opinion :D.

File: config/config-builder/src/full_node_config.rs [hunk 3]
```diff
@@ -218,6 +221,8 @@ impl FullNodeConfig {
                 .full_node_networks
                 .get_mut(0)
                 .ok_or(Error::MissingFullNodeNetwork)?;
+            // @TODO: We need to determine a name for this Private network
```
Comment (davidiw) [inline config/config-builder/src/full_node_config.rs:224]: this is config-builder... if this is a validator node or a fullnode connected to a validator, then this should probably be "vfn"
otherwise this is the public network

let's just call all the networks in config-builder private "vfn" and let @mgorven help us adjust if possible.

I think @agclement has also spent time here, perhaps he can make a suggestion?

File: config/management/src/smoke_test.rs [hunk 4]
```diff
@@ -71,10 +72,12 @@ fn smoke_test() {
         let mut network = NetworkConfig::default();
         network.discovery_method = DiscoveryMethod::Onchain;
         network.peer_id = validator_account;
+        network.network_id = NetworkId::validator_network();
         config.validator_network = Some(network);
 
         let mut network = NetworkConfig::default();
         network.discovery_method = DiscoveryMethod::None;
+        network.network_id = NetworkId::public_network();
```
Comment (davidiw) [inline config/management/src/smoke_test.rs:80]: this is private_network "vfn"

File: config/src/config/mod.rs [hunk 5]
```diff
@@ -278,7 +290,9 @@ impl NodeConfig {
             );
 
             if self.validator_network.is_none() {
-                self.validator_network = Some(NetworkConfig::default());
+                let mut network_config = NetworkConfig::default();
+                network_config.network_id = NetworkId::validator_network();
```
Comment (davidiw) [inline config/src/config/mod.rs:294]: maybe we should make a new function that takes in a network id? maybe not...

File: config/src/config/test_data/random.complete.node.config.toml [hunk 6]
```diff
@@ -82,3 +82,7 @@ enable_remote_authentication = true
 network_peers_file = ""
 seed_peers_file = "31893204fa402143c11b26ce8a89ea1d.seed_peers.toml"
 identity_private_key = "60dd107034b4582a2ef42c5e1ea475f2fea477a10a9f1d75b3635243b2506b72"
+
+[validator_network.network_id]
+network_name = ""
```
Comment (davidiw) [inline config/src/config/test_data/random.complete.node.config.toml:87]: this shouldn't have a network_name :O
also I would just use name and type here otherwise it is a bit redundant

File: config/src/network_id.rs [hunk 7]
```diff
@@ -0,0 +1,87 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub enum NetworkType {
+    Validator = 0,
+    Public,
+    Private,
```
Comment (davidiw) [inline config/src/network_id.rs:9]: Why didn't we follow the proposed format of

```
pub enum NetworkType {
  Validator,
  Public,
  Private(PrivateNetwork),
}

pub struct PrivateNetwork {
  name: String,
}
```

File: config/src/network_id.rs [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        }
+    }
+
+    /// Global Validator network
+    pub fn validator_network() -> NetworkId {
+        NetworkId {
+            network_type: NetworkType::Validator,
+            network_name: "".to_string(),
+        }
+    }
```
Comment (davidiw) [inline config/src/network_id.rs:62]: your current approach requires this extra code.

File: network/src/transport.rs [hunk 9]
```diff
@@ -397,3 +420,151 @@ pub fn build_tcp_transport(
         .with_timeout(TRANSPORT_TIMEOUT)
         .boxed()
 }
+
+#[cfg(test)]
```
Comment (davidiw) [inline network/src/transport.rs:424]: feel mixed about these tests, I don't know if we benefit from adding them to the code base, it is a lot to read and potentially modify... Ideally we cover these from a unit test perspective in the HandshakeMsg. Then we also handle it from our various integration and smoke tests. Thoughts? 150 lines of test code is a lot. Question, did this help you develop it?

File: config/config-builder/src/full_node_config.rs (hunk 2 above)
Comment (mgorven) [reply config/config-builder/src/full_node_config.rs:171]: If public==unauthenticated then yes, but there may also be an authenticated fullnode-only network for trusted external fullnodes to use (e.g. fullnodes run by Novi in FB infra connecting to the fullnode of Novi's validator deployment).

File: config/config-builder/src/full_node_config.rs (hunk 3 above)
Comment (mgorven) [reply config/config-builder/src/full_node_config.rs:224]: That sounds good enough for now.

File: config/config-builder/src/full_node_config.rs (hunk 2 above)
Comment (agc-quadra) [reply config/config-builder/src/full_node_config.rs:171]: TL;DR --- the term _public_ is overloaded and _public network_ means different things to different people.

TL;DR(2) --- A clear, concise, consistent understanding of deployment targets is needed.


A couple of different definitions of _public_

* unauthenticated and open to full participation by anyone
* accessible to anyone

The former definition is most inline with traditional blockchain projects.  This definition goes hand in hand with unconstrained deployments.

The second definition allows for networks which are deployed in a controlled environment, but accessible to anyone. A reasonable analogue is the public front ends that stand between users  and the back end infrastructure.

Looking at the [discussion](https://github.com/libra/libra/issues/3985#issuecomment-636659952) in #3985, one irony is that the suggested "private" networks connected to the public ring are in fact the most inline with the first definition of public.

File: config/src/config/test_data/random.complete.node.config.toml (hunk 6 above)
Comment (gregnazario) [reply config/src/config/test_data/random.complete.node.config.toml:87]: `type` is a reserved word in rust :(. So, to be consistent I put `network` before both.

File: config/src/network_id.rs (hunk 7 above)
Comment (gregnazario) [reply config/src/network_id.rs:9]: TOML serialization determines that an ENUM with any extra values was an UnsupportedType.  See: https://github.com/libra/libra/issues/4210

I can go back and try it, but this should get the ball rolling.

File: config/src/network_id.rs (hunk 8 above)
Comment (gregnazario) [reply config/src/network_id.rs:62]: I can cut this out, the reason it required extra code was I found myself constantly switching between enums and structs trying to figure out which one was actually allowed in TOML.  By abstracting it away it let me change it easier.

I really wanted to use the enum approach, but TOML caused a whole bunch of serialization issues.

File: network/src/transport.rs (hunk 9 above)
Comment (gregnazario) [reply network/src/transport.rs:424]: It's a lot because I copied them from a different class mostly because I didn't know what I was doing.

The issue here is there are no tests actually checking if there are checks on the handshakes.  All of the tests are whether the handshake works at a wire level, but there are none at the transport level.

I think a negative test is important to know that we are actually validating our connections.  The smoke test can handle the positive test.

File: config/src/config/mod.rs (hunk 5 above)
Comment (gregnazario) [reply config/src/config/mod.rs:294]: I can take a look into that, there's just a lot of defaults used, and I think when I started this I wasn't sure if the higher level configs call NetworkConfig default.

File: config/src/network_id.rs (hunk 7 above)
Comment (gregnazario) [reply config/src/network_id.rs:9]: For everyone's knowledge, toml, won't serialize the enum:
- toml: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0e206f00f9d42117e9c3dc1906ba1762
- yaml: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0e206f00f9d42117e9c3dc1906ba1762

Comment (gregnazario) [reply config/src/network_id.rs]: Got it mostly working after putting in a tag.  Now just some side effects in the serde analyzer.

File: config/src/config/test_data/random.complete.node.config.toml (hunk 6 above)
Comment (gregnazario) [reply config/src/config/test_data/random.complete.node.config.toml]: Solved with a serde tag.

File: config/src/config/mod.rs (hunk 5 above)
Comment (gregnazario) [reply config/src/config/mod.rs]: I've combined these.  But, can't fully get rid of `NetworkConfig::default()` because it is used in deserialization.

File: config/src/network_id.rs [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+pub struct NetworkInfo {
+    name: String,
+}
+
+/// A representation of the network being used in communication.
+/// There should only be one of each NetworkId used for a single node, and handshakes should verify
+/// that the NetworkId being used is the same during a handshake, to effectively ensure communication
+/// is restricted to a network.  Network should be checked that it is not the `DEFAULT_NETWORK`
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[serde(tag = "type")]
```
Comment (matbd) [inline config/src/network_id.rs:15]: This option is likely the cause of the breakage in the `generate-format` unit test. It would not work with LCS in first place.

Any reason for using this? Is this meant to be serialized both in JSON and LCS? (then I don't have a solution but to accept a slightly uglier JSON)

Comment (davidiw) [reply config/src/network_id.rs:15]: Can we eliminate the notion of "tag" from LCS, I don't think we need to serialize / deserialize this field... I don't know why serde would do this to us :P

Comment (matbd) [reply config/src/network_id.rs:15]: OK so if we really want to share the definition of `NetworkId` between TOML and LCS, here is a solution:
```
#[derive(Clone, Debug, Eq, Hash, PartialEq)]
pub enum NetworkId {
    Validator,
    Public,
    Private(NetworkInfo),
}

#[derive(Serialize, Deserialize)]
#[serde(tag = "type", rename = "NetworkId")]
enum HumanReadableNetworkId {
    Validator,
    Public,
    Private(NetworkInfo),
}

#[derive(Serialize, Deserialize)]
#[serde(rename = "NetworkId")]
enum NonHumanReadableNetworkId {
    Validator,
    Public,
    Private(NetworkInfo),
}

impl Serialize for NetworkId {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer
    {
        if serializer.is_human_readable() {
            match self {
                NetworkId::Validator => HumanReadableNetworkId::Validator.serialize(serializer),
                NetworkId::Public => HumanReadableNetworkId::Public.serialize(serializer),
                NetworkId::Private(info) => HumanReadableNetworkId::Private(*info).serialize(serializer),
            }
        } else {
            match self {
                NetworkId::Validator => NonHumanReadableNetworkId::Validator.serialize(serializer),
                NetworkId::Public => NonHumanReadableNetworkId::Public.serialize(serializer),
                NetworkId::Private(info) => NonHumanReadableNetworkId::Private(*info).serialize(serializer),
            }
        }
    }
}

impl<'de> Deserialize<'de> for NetworkId {

    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>
    {
        if deserializer.is_human_readable() {
            match HumanReadableNetworkId::deserialize(deserializer)? {
                HumanReadableNetworkId::Validator => Ok(NetworkId::Validator),
                HumanReadableNetworkId::Public => Ok(NetworkId::Public),
                HumanReadableNetworkId::Private(info) => Ok(NetworkId::Private(info)),
            }
        } else {
            match NonHumanReadableNetworkId::deserialize(deserializer)? {
                NonHumanReadableNetworkId::Validator => Ok(NetworkId::Validator),
                NonHumanReadableNetworkId::Public => Ok(NetworkId::Public),
                NonHumanReadableNetworkId::Private(info) => Ok(NetworkId::Private(info)),
            }
        }
}
```

Obviously, it's not ideal but it works. If this happens often in the future, we could think of some procedural macros to help.

File: config/src/config/mod.rs [hunk 11]
```diff
@@ -287,7 +300,8 @@ impl NodeConfig {
         } else {
             self.validator_network = None;
             if self.full_node_networks.is_empty() {
-                self.full_node_networks.push(NetworkConfig::default());
+                let network_config = NetworkConfig::network_with_id(NetworkId::Public);
```
Comment (davidiw) [inline config/src/config/mod.rs:303]: @mgorven is this the public network or is it the internal network for VFN or are we even going to use config-builder at all come launch?

Comment (davidiw) [reply config/src/config/mod.rs:303]: note: this is the FN directly on the validator

File: config/src/config/mod.rs [hunk 12]
```diff
@@ -400,7 +414,6 @@ mod test {
         let root_dir = RootPath::new(expected.test.as_ref().unwrap().temp_dir().unwrap());
         let path = root_dir.full_path(&PathBuf::from("node.config.toml"));
         expected.save(&path).expect("Unable to save config");
-
```
Comment (davidiw) [inline config/src/config/mod.rs:403]: why do you hate whitespace?

File: config/src/network_id.rs [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+/// that the NetworkId being used is the same during a handshake, to effectively ensure communication
+/// is restricted to a network.  Network should be checked that it is not the `DEFAULT_NETWORK`
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum NetworkId {
+    Validator,
+    Public,
+    Private(NetworkInfo),
+}
+
+const INVALID: &str = "Invalid";
```
Comment (davidiw) [inline config/src/network_id.rs:21]: why?

File: config/src/network_id.rs [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+impl NetworkId {
+    /// Tell us whether the `NetworkId` was configured correctly.
+    /// Private networks must have a name, otherwise we can't differentiate.
+    pub fn is_valid(&self) -> bool {
+        match self {
+            NetworkId::Validator => true,
+            NetworkId::Public => true,
+            NetworkId::Private(network_info) => network_info.name == INVALID,
+        }
+    }
```
Comment (davidiw) [inline config/src/network_id.rs:39]: they should all be valid imo, not sure what we get by having this?

File: config/src/network_id.rs [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename = "NetworkId")]
+enum NonHumanReadableNetworkId {
+    Validator,
+    Public,
+    Private(NetworkInfo),
+}
+
+impl Serialize for NetworkId {
```
Comment (davidiw) [inline config/src/network_id.rs:68]: very clever guys, thanks @matbd

File: network/src/transport.rs [hunk 16]
```diff
@@ -397,3 +420,92 @@ pub fn build_tcp_transport(
...
+        let public_network = NetworkId::Public;
+
+        // Create client and server handshake messages.
+        let mut server_handshake = HandshakeMsg::new(validator_network);
+        server_handshake.add(
+            MessagingProtocolVersion::V1,
+            [
+                ProtocolId::ConsensusDirectSend,
+                ProtocolId::MempoolDirectSend,
+            ]
```
Comment (davidiw) [inline network/src/transport.rs:459]: we don't need to actually be this verbose, so maybe we can just strip out a lot of boiler plate stuff like this?

File: network/src/transport.rs [hunk 17]
```diff
@@ -397,3 +420,92 @@ pub fn build_tcp_transport(
...
+            )
+            .await
+            .expect_err(
+                format!(
+                    "network_ids don't match own: {:?} received: {:?}",
+                    NetworkId::Validator,
+                    NetworkId::Public
+                )
+                .as_str(),
+            );
```
Comment (davidiw) [inline network/src/transport.rs:487]: probably could just do an unwrap_err here, unless you think we get additional signal from this? I don't think we do

File: network/src/transport.rs [hunk 18]
```diff
@@ -397,3 +420,92 @@ pub fn build_tcp_transport(
...
+        let client = async move {
+            perform_handshake(
+                peer_id_2,
+                outbound,
+                network_addr_2,
+                ConnectionOrigin::Outbound,
+                &client_handshake,
+            )
+            .await
+            .expect_err(
```
Comment (davidiw) [inline network/src/transport.rs:499]: same a simple unwrap_err will strip out a lot of code

File: config/src/config/mod.rs (hunk 11 above)
Comment (mgorven) [reply config/src/config/mod.rs:303]: I'm not sure when this code is run, but I'm guessing this is the internal validator-fullnode network on the fullnode side. For launch we need fullnodes to have two networks: the validator-fullnode network, and the public unauthenticated network.

I would prefer not to use `config-builder` in production, but we need a way to handle multiple fullnodes on the validator-fullnode network. If the fullnodes can all use the `fullnode_network` key from Vault with random peer IDs when talking to the validator then that should work. We just need to figure out the validator-side of that to trust that `fullnode_network` key.

Comment (davidiw) [reply config/src/config/mod.rs:303]: Let's talk offline about this and come up with a scheme that works. Networking is complicated :/

File: config/src/config/mod.rs (hunk 12 above)
Comment (gregnazario) [reply config/src/config/mod.rs:403]: I actually really like whitespace.

I spent a lot of time putting printlns places to see what was going on, I may have deleted this line.

File: config/src/network_id.rs (hunk 13 above)
Comment (gregnazario) [reply config/src/network_id.rs:21]: I don't think this necessarily has to be a constant.

The purpose of this is when deserializing, it seemed I needed a default value (probably for the struct that I had before).  So, I added this to ensure that if it had been loaded with a lack of a configuration for a network, it wouldn't just all of a sudden connect to one of the three networks.

File: config/src/network_id.rs (hunk 14 above)
Comment (gregnazario) [reply config/src/network_id.rs:39]: Same as above.  I'll check if the Default value was because of the struct or an enum.

File: network/src/transport.rs (hunk 16 above)
Comment (gregnazario) [reply network/src/transport.rs:459]: I stole this all from https://github.com/libra/libra/blob/1df4359a5fc4d2c82b636152987071b57021c768/network/src/protocols/identity.rs#L89-L138

I'll take a round to simplify it.

The only reason I added the test here and not in the identity protocol, is because it doesn't cover the application level check on equality of networkIds.

File: config/src/network_id.rs (hunk 14 above)
Comment (gregnazario) [reply config/src/network_id.rs:39]: The purpose of this was that there is a `Default` for `NetworkConfig` when deserializing.  However, I wanted to be explicit that if `NetworkConfig` fills in the default value, that we should not trust it because it loaded a default value from the `NetworkConfig`.

I'm not sure having a defaulting configuration is really what we want on a `NetworkConfig`

File: network/src/transport.rs (hunk 17 above)
Comment (gregnazario) [reply network/src/transport.rs:487]: I think the idea was to ensure we aren't just getting some random other error

File: config/src/network_id.rs [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+pub enum NetworkId {
+    Validator,
+    Public,
+    Private(NetworkInfo),
+}
+
+/// Default needed to handle downstream structs that use `Default`
+impl Default for NetworkId {
+    fn default() -> NetworkId {
+        NetworkId::private_network("Default")
```
Comment (davidiw) [inline config/src/network_id.rs:24]: let's make the default the Public network or Validator network.

**Comment by davidiw** [comment]
Don't forget to rebase!

**Comment by gregnazario** [comment]
> Don't forget to rebase!

Let's hope that the 3rd rebase is the charm
